---
title: "meth_vs_distance"
author: "Chenxin Li"
date: "April 6, 2019"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(stringr)
library(svglite)
library(rsq)
```

```{r}
egg_specific <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS08/loci gene distances/egg-specific_closest_gene.txt", header=FALSE)
```
```{r}
sd_specific <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS08/loci gene distances/seedling-specific_closest_gene.txt", header=FALSE)
```
```{r}
sp_specific <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS08/loci gene distances/sperm-specific_closest_gene.txt", header=FALSE)
```
```{r}
intersect <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS08/loci gene distances/intersection_closest_gene.txt", header=FALSE)
```

```{r}
egg_specific_df <- egg_specific %>% 
  select(V1, V4, V22) %>% 
  mutate(bin = "egg-specific") 

```
```{r}
sd_specific_df <- sd_specific %>% 
  select(V1, V4, V22) %>% 
  mutate(bin = "seedling-specific") 
```
```{r}
sp_specific_df <- sp_specific %>% 
  select(V1, V4, V22) %>% 
  mutate(bin = "sperm-specific") 
```
```{r}
intersect_df <- intersect %>% 
  select(V1, V4, V22) %>% 
  mutate(bin = "intersect.") 
```
```{r}
distance_to_gene <- rbind(egg_specific_df,
                          sd_specific_df,
                          sp_specific_df,
                          intersect_df) 

colnames(distance_to_gene) <- c("chr", "start", "distance", "bin")
```

#Use leaf as training data
##load data
```{r}
wt_in_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503136-7.trimmed.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt_in_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503136-7.trimmed.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt_in_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503136-7.trimmed.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt_in_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503136-7.trimmed.fastq.gz.sorted.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)
```
##tidy data
```{r}
clean <- function(df){
  df %>% 
    select(X1, X4, X13, X14, X15, X21, X22) %>% 
    filter(X21 > 0.9) %>% 
    filter(X22 >= 3) %>% 
    select(-X21, -X22)
}
```

```{r}
wt_in_egg1 <- wt_in_egg %>% 
  clean() %>% 
  mutate(sample_ID = "WT") %>% 
  mutate(bin = "egg-specific") %>% 
  mutate(genotype = "WT")

colnames(wt_in_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")

wt_in_sd1 <- wt_in_sd %>% 
  clean() %>% 
  mutate(sample_ID = "WT") %>% 
  mutate(bin = "seedling-specific") %>% 
  mutate(genotype = "WT")

colnames(wt_in_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")


wt_in_sp1 <- wt_in_sp %>% 
  clean() %>% 
  mutate(sample_ID = "WT") %>% 
  mutate(bin = "sperm-specific") %>% 
  mutate(genotype = "WT")


colnames(wt_in_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")

wt_in_inter1 <- wt_in_inter %>% 
  clean() %>% 
  mutate(sample_ID = "WT") %>% 
  mutate(bin = "intersect.") %>% 
  mutate(genotype = "WT")

colnames(wt_in_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
```
```{r}
wt_leaf <- rbind(wt_in_egg1, 
                 wt_in_sd1,
                 wt_in_sp1,
                 wt_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>% 
  filter(distance > 100) 
```
```{r}
quantile(wt_leaf$distance, 0.95)
quantile(wt_leaf$distance, 0.95)/20
```
`

```{r}
wt_leaf_s <- rbind(wt_in_egg1, 
                 wt_in_sd1,
                 wt_in_sp1,
                 wt_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>%
  filter(distance > 100) %>% 
  mutate(bucket = ntile(distance, 750)) %>% 
  gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context) %>% 
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()

head(wt_leaf_s, 100)
```


```{r}
wt_leaf_s %>% 
  ggplot(aes(x = mean.distance, y = mean.methylation)) +
  facet_grid(context ~ . , switch = "y", scales = "free_y") +
  # geom_point(data = wt_leaf %>% 
  #              filter(distance < quantile(wt_leaf$distance, 0.95)) %>% 
  #              sample_n(1000) %>% 
  #              gather("context", "methylation", 3:5), aes(x = distance, y = methylation)) +
  geom_smooth() +
  geom_point(size = 0.5, alpha = 0.8, color = "grey20") +
  geom_vline(xintercept = 700, linetype = 4, size = 1, color = "red") +
  labs(y = NULL,
       x = "distance to nearest gene/bp") +
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

```


```{r}
model_wt_leaf_chh <- glm(mean.methylation ~ mean.distance, family = binomial(link = "inverse"), wt_leaf_s %>%
                           filter(context == "mCHH"))
```




```{r}
summary(model_wt_leaf_chh)
model_wt_leaf_null <- glm(mean.methylation ~ 1, family = binomial(link = "inverse"), wt_leaf_s %>% 
                            filter(context == "mCHH"))
anova(model_wt_leaf_chh)
```
```{r}
leaf_wt_chh_s <- summary(model_wt_leaf_chh)
leaf_wt_chh_coef <- leaf_wt_chh_s$coefficients
leaf_wt_chh_coef
```

```{r}
logit <- function(p){log(
  p / (1-p)
)}
invlogit <- function(x){
  1/(1 + exp(-x))
}
```
```{r}
wt_leaf_chh_fitted <- data.frame("distance" = seq(0, 20000, by = 100)) %>% 
  mutate(predicted_mCHH = 1/(
    (leaf_wt_chh_coef[2, 1] * distance + leaf_wt_chh_coef[1, 1])
    ))

wt_leaf_chh_fitted %>% head(10)
```
```{r}
rsq(model_wt_leaf_chh)
```

##piece-wise regression
```{r}
wt_leaf_near_s <- rbind(wt_in_egg1, 
                 wt_in_sd1,
                 wt_in_sp1,
                 wt_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance > 100) %>% 
  filter(distance < 1000) %>% 
  mutate(bucket = ntile(distance, 50)) %>% 
  gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context) %>% 
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()

dim(wt_leaf_near_s)
```

```{r}
wt_leaf_near_s %>% 
  ggplot(aes(x = mean.distance, y = mean.methylation)) +
  facet_grid(context ~ . , switch = "y", scales = "free_y") +
  # geom_point(data = wt_leaf %>% 
  #              filter(distance < quantile(wt_leaf$distance, 0.95)) %>% 
  #              sample_n(1000) %>% 
  #              gather("context", "methylation", 3:5), aes(x = distance, y = methylation)) +
  geom_smooth() +
  geom_point(size = 0.5, alpha = 0.8, color = "grey20") +
  labs(y = NULL,
       x = "distance to nearest gene/bp") +
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))
```
```{r}
model_wt_leaf_chh_near <- lm(mean.methylation %>% logit() ~ mean.distance, wt_leaf_near_s %>%
                           filter(context == "mCHH"))
plot(model_wt_leaf_chh_near)
```
```{r}
summary(model_wt_leaf_chh_near)
wt_leaf_chh_near_coef <- summary(model_wt_leaf_chh_near)$coef
```
```{r}
wt_leaf_chh_fitted_near <- data.frame("distance" = seq(0, 1500)) %>% 
  mutate(predicted_mCHH = invlogit(
    (wt_leaf_chh_near_coef[2, 1] * distance + wt_leaf_chh_near_coef[1, 1])
    ))

wt_leaf_chh_fitted %>% head(10)
```

#in drm2 leaf
```{r}
drm_in_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503142.trimmed.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_in_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503142.trimmed.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_in_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503142.trimmed.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_in_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503142.trimmed.fastq.gz.sorted.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)
```
```{r}
drm_in_egg1 <- drm_in_egg %>% 
  clean() %>% 
  mutate(sample_ID = "drm2") %>% 
  mutate(bin = "egg-specific") %>% 
  mutate(genotype = "drm2")
colnames(drm_in_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
drm_in_sd1 <- drm_in_sd %>% 
  clean() %>% 
  mutate(sample_ID = "drm2") %>% 
  mutate(bin = "seedling-specific") %>% 
  mutate(genotype = "drm2")
colnames(drm_in_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
drm_in_sp1 <- drm_in_sp %>% 
  clean() %>% 
  mutate(sample_ID = "drm2") %>% 
  mutate(bin = "sperm-specific") %>% 
  mutate(genotype = "drm2")
colnames(drm_in_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
drm_in_inter1 <- drm_in_inter %>% 
  clean() %>% 
  mutate(sample_ID = "drm2") %>% 
  mutate(bin = "intersect.") %>% 
  mutate(genotype = "drm2")
colnames(drm_in_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
```

```{r}
drm2_leaf_s <- rbind(drm_in_egg1, 
                 drm_in_sd1,
                 drm_in_sp1,
                 drm_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>%
  filter(distance > 100) %>% 
  mutate(bucket = ntile(distance, 750)) %>% 
  gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context) %>% 
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()

head(drm2_leaf_s, 100)
```

#ddm1
```{r}
ddm_in_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503134.trimmed.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

ddm_in_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503134.trimmed.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

ddm_in_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503134.trimmed.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

ddm_in_inter  <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/SRR3503134.trimmed.fastq.gz.sorted.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1) 
```

```{r}
ddm_in_egg1 <- ddm_in_egg %>% 
  clean() %>% 
  mutate(sample_ID = "ddmab") %>% 
  mutate(bin = "egg-specific") %>% 
  mutate(genotype = "ddm1ab")
colnames(ddm_in_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
ddm_in_sd1 <- ddm_in_sd %>% 
  clean() %>% 
  mutate(sample_ID = "ddmab") %>% 
  mutate(bin = "seedling-specific") %>% 
  mutate(genotype = "ddm1ab")
colnames(ddm_in_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
ddm_in_sp1 <- ddm_in_sp %>% 
  clean() %>% 
  mutate(sample_ID = "ddmab") %>% 
  mutate(bin = "sperm-specific") %>% 
  mutate(genotype = "ddm1ab")
colnames(ddm_in_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
ddm_in_inter1 <- ddm_in_inter %>%
  clean() %>% 
  mutate(sample_ID = "ddmab") %>% 
  mutate(bin = "intersect.") %>% 
  mutate(genotype = "ddm1ab")
colnames(ddm_in_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin", "genotype")
```


```{r}
ddm1_leaf_s <- rbind(ddm_in_egg1, 
                 ddm_in_sd1,
                 ddm_in_sp1,
                 ddm_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>%
  filter(distance > 100) %>% 
  mutate(bucket = ntile(distance, 750)) %>% 
  gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context) %>% 
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()

head(ddm1_leaf_s, 100)
```
```{r}
ggplot(wt_leaf_chh_fitted, aes(x = distance, y = predicted_mCHH)) +
  geom_point(data = wt_leaf_s %>% 
               filter(context == "mCHH"), 
               aes(x = mean.distance, y = mean.methylation, color = "obvserved values WT leaf"), size = 1, alpha = 0.5) +
  geom_point(data = ddm1_leaf_s %>% 
               filter(context == "mCHH"), 
               aes(x = mean.distance, y = mean.methylation, color = "obvserved values ddm1 leaf"), size = 1, alpha = 0.5) +
  geom_point(data = drm2_leaf_s %>% 
               filter(context == "mCHH"), 
               aes(x = mean.distance, y = mean.methylation, color = "obvserved values drm2 leaf"), size = 1, alpha = 0.5) +
  geom_line(aes(color = "global regression WT leaf"), size = 2, alpha =  0.8, lineend = "round") +
  geom_line(data = wt_leaf_chh_fitted_near, 
            aes(color = "near gene regression WT leaf"), size = 2, alpha = 0.8, lineend = "round") + 
  scale_color_manual(values = c("darkgreen", "green2", "red", "blue","black")) + 
  labs(y = "mCHH", 
       x = "distance to nearest gene/bp",
       color = NULL) +
  theme_minimal() +
  theme(legend.position = c(0.6, 0.85)) +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("leaf_regression.svg", height = 5, width = 5)
```
WT regression lines are unfit for ddm1ab and drm2 leaf for raw methylation values

#gametes
```{r}
egg_in_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/egg100.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.nondir6.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

egg_in_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/egg100.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.nondir6.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

egg_in_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/egg100.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.nondir6.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

egg_in_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/egg100.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.nondir6.sorted.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)
```

```{r}
egg_in_egg1 <- clean(egg_in_egg) %>% 
  mutate(sample_ID = "egg100") %>% 
  mutate(bin = "egg-specific")

colnames(egg_in_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")


egg_in_sd1 <- clean(egg_in_sd) %>% 
  mutate(sample_ID = "egg100") %>% 
  mutate(bin = "seedling-specific")

colnames(egg_in_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")

egg_in_sp1 <- clean(egg_in_sp) %>% 
  mutate(sample_ID = "egg100") %>% 
  mutate(bin = "sperm-specific")

colnames(egg_in_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")

egg_in_inter1 <- clean(egg_in_inter) %>% 
  mutate(sample_ID = "egg100") %>% 
  mutate(bin = "intersect.")

colnames(egg_in_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
egg_s <- rbind(egg_in_egg1, 
                 egg_in_sd1,
                 egg_in_sp1,
                 egg_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>%
  filter(distance > 100) %>% 
  mutate(bucket = ntile(distance, 750)) %>% 
  gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context) %>% 
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()

head(egg_s, 100)
```
```{r}
sperm_in_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/sperm1.split.nondir6.sorted.merged.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

sperm_in_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/sperm1.split.nondir6.sorted.merged.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

sperm_in_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/sperm1.split.nondir6.sorted.merged.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

sperm_in_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/sperm1.split.nondir6.sorted.merged.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)
```


```{r}
sperm_in_egg1 <- clean(sperm_in_egg) %>% 
  mutate(sample_ID = "sperm1") %>% 
  mutate(bin = "egg-specific")
colnames(sperm_in_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")

sperm_in_sd1 <- clean(sperm_in_sd) %>% 
  mutate(sample_ID = "sperm1") %>% 
  mutate(bin = "seedling-specific")
colnames(sperm_in_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")

sperm_in_sp1 <- clean(sperm_in_sp) %>% 
  mutate(sample_ID = "sperm1") %>% 
  mutate(bin = "sperm-specific")
colnames(sperm_in_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")

sperm_in_inter1 <- clean(sperm_in_inter) %>% 
  mutate(sample_ID = "sperm1") %>% 
  mutate(bin = "intersect.")
colnames(sperm_in_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
sperm_s <- rbind(sperm_in_egg1, 
                 sperm_in_sd1,
                 sperm_in_sp1,
                 sperm_in_inter1) %>% 
  full_join(distance_to_gene, by = c("chr", "start", "bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>%
  filter(distance > 100) %>% 
  mutate(bucket = ntile(distance, 750)) %>% 
  gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context) %>% 
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()

head(sperm_s, 100)
```

```{r}
obs_values <- wt_leaf_s %>% 
  mutate(sample_type = "leaf") %>%
  rbind(egg_s %>% 
          mutate(sample_type = "egg")) %>% 
  rbind(sperm_s %>% 
          mutate(sample_type = "sperm")) %>% 
  filter(context == "mCHH") %>% 
  group_by(sample_type) %>% 
  mutate(z.score = (mean.methylation - mean(mean.methylation))/sd(mean.methylation)) %>% 
  ungroup()
```
##fit model for mCHH z-score in leaf - use z-score to reduce systematic bias due to different library prep methods 
```{r}
model_z <- lm(z.score ~ I(mean.distance) + I(mean.distance^2), obs_values %>% 
                filter(sample_type == "leaf"))

summary(model_z)
plot(model_z)
anova(model_z)
```
```{r}
global_coef <- summary(model_z)$coef 
global_coef
```

```{r}
wt_leaf_z_fitted <- data.frame("distance" = seq(0, 20000, by = 100)) %>% 
  mutate(predicted_z = global_coef[3, 1] * distance^2 + global_coef[2, 1] * distance + global_coef[1, 1])

wt_leaf_z_fitted %>% head(10)
```
```{r}
model_z_near <- lm(z.score ~ mean.distance, obs_values %>% 
                filter(sample_type == "leaf") %>% 
                  filter(mean.distance < 750) %>% 
                  filter(mean.distance > 100))

summary(model_z_near)
plot(model_z_near)
anova(model_z_near)
```
```{r}
near_coef <- summary(model_z_near)$coef 
near_coef
```
```{r}
wt_leaf_z_fitted_near <- data.frame("distance" = seq(0, 1000)) %>% 
  mutate(predicted_z = near_coef[2, 1] * distance + near_coef[1, 1])

wt_leaf_z_fitted_near %>% head(10)
```
```{r}
predicted_values <- wt_leaf_z_fitted %>% 
  mutate(method = "global regression WT leaf") %>% 
  rbind(wt_leaf_z_fitted_near %>% 
          mutate(method = "near gene regression WT leaf")) 

head(predicted_values)
```


```{r}
predicted_values %>% 
ggplot(aes(x = distance, y = predicted_z)) +
  geom_point(data = obs_values,
               aes(x = mean.distance,
                   y = z.score,
                   color = sample_type), size = 1, alpha = 0.5) +
  geom_line(aes(group = method, color = method), size = 2, alpha =  0.8, lineend = "round") +
  scale_color_manual(values = c("tomato1", "darkgreen", "black", "green2", "dodgerblue2")) + 
  labs(y = "mCHH z-score", 
       x = "distance to nearest gene/bp",
       color = NULL) +
  theme_minimal() +
  theme(legend.position = c(0.6, 0.85)) +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("leaf_vs_gamete_regression_dots.svg", height = 5, width = 5)
```

#goodness of fit for gamates
sum((o - e)^2)/df
```{r}
obs_values <- obs_values %>% 
  mutate(predicted.z.near = near_coef[2, 1] * mean.distance + near_coef[1, 1]) %>% 
  mutate(predicted.z.global = global_coef[3, 1] * mean.distance^2 + global_coef[2, 1] * mean.distance + global_coef[1, 1]) %>% 
  mutate(xsq.global = (z.score - predicted.z.global)^2 ) %>% 
  mutate(xsq.near = (z.score - predicted.z.near)^2) 


```
```{r}
obs_leaf <- filter(obs_values, sample_type == "leaf") %>% mutate(df = n() - 1)
obs_egg <- filter(obs_values, sample_type == "egg") %>% mutate(df = n() - 1)
obs_sperm <- filter(obs_values, sample_type == "sperm") %>% mutate(df = n() - 1)
```

```{r}
cor.test(obs_leaf$predicted.z.global, obs_leaf$z.score)

cor.test(obs_egg$predicted.z.global, obs_leaf$z.score)

cor.test(obs_sperm$predicted.z.global, obs_leaf$z.score)
```
 


```{r}
library(boot)
f.sum <- function(Y,i) sum(Y[i])/(750 - 3)
```
```{r}
alpha <- 0.05/choose(3, 2)
```

```{r}
egg_ssq <- boot(obs_egg$xsq.global, R = 1000, statistic = f.sum)
egg_quantiles <- quantile(egg_ssq$t, c(alpha/2, 1-alpha/2))
```
```{r}
leaf_ssq <- boot(obs_leaf$xsq.global, R = 1000, statistic = f.sum)
leaf_quantiles <- quantile(leaf_ssq$t, c(alpha/2, 1-alpha/2))
```
```{r}
sperm_ssq <- boot(obs_sperm$xsq.global, R = 1000, statistic = f.sum)
sperm_quantiles <- quantile(sperm_ssq$t, c(alpha/2, 1-alpha/2))
```



```{r}
msq <- c(
  obs_egg %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747,
  obs_leaf %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747,
  obs_sperm %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747
)

sample_type <- c("egg", "leaf", "sperm")


```


```{r}
msq.global <- data.frame(sample_type, msq) %>% 
  mutate(method = "global") %>% 
  cbind(rbind(egg_quantiles, leaf_quantiles,sperm_quantiles))

msq.global
```
```{r}
obs_leaf_near <- filter(obs_values, sample_type == "leaf") %>% 
  filter(mean.distance < 750) %>% 
  mutate(df = n() - 1)

obs_egg_near <- filter(obs_values, sample_type == "egg") %>% 
  filter(mean.distance < 750) %>% 
  mutate(df = n() - 1)

obs_sperm_near <- filter(obs_values, sample_type == "sperm") %>% 
  filter(mean.distance < 750) %>% 
  mutate(df = n() - 1)
```

```{r}
cor.test(obs_leaf_near$z.score, obs_leaf_near$predicted.z.near)

cor.test(obs_egg_near$z.score, obs_egg_near$predicted.z.near)

cor.test(obs_sperm_near$z.score, obs_sperm_near$predicted.z.near)
```
```{r}
obs_values %>% 
  filter(mean.distance < 750) %>% 
ggplot(aes(x = z.score, y = predicted.z.near)) +
  geom_point(aes(color = sample_type)) +
  geom_abline(slope = 1, intercept = 0) 
```

```{r}
msq_near <- c(
  obs_egg_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133,
  obs_leaf_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133,
  obs_sperm_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133
)

sample_type <- c("egg", "leaf", "sperm")
```


```{r}
f.sum_near <- function(Y,i) sum(Y[i])/133
```

```{r}
egg_near_ssq <- boot(obs_egg_near$xsq.near, R = 1000, statistic = f.sum_near)
egg_near_quantiles <- quantile(egg_near_ssq$t, c(alpha/2, 1-alpha/2))

leaf_near_ssq <- boot(obs_leaf_near$xsq.near, R = 1000, statistic = f.sum_near)
leaf_near_quantiles <- quantile(leaf_near_ssq$t, c(alpha/2, 1-alpha/2))

sperm_near_ssq <- boot(obs_sperm_near$xsq.near, R = 1000, statistic = f.sum_near)
sperm_near_quantiles <- quantile(sperm_near_ssq$t, c(alpha/2, 1-alpha/2))
```
```{r}
msq.near <- data.frame(sample_type, msq_near) %>% 
  mutate(method = "near gene") %>% 
  cbind(rbind(egg_near_quantiles, leaf_near_quantiles,sperm_near_quantiles))

colnames(msq.near)[2] <- "msq"
msq.near
```
```{r}
msq.summary <- rbind(msq.global, msq.near)

colnames(msq.summary)[4:5] <- c("lower.CL", "upper.CL")

msq.summary 
```
```{r}
egg_msq <- egg_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "egg") %>% 
  mutate(method = "global")

leaf_msq <- leaf_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "leaf") %>% 
  mutate(method = "global")

sperm_msq <- sperm_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "sperm") %>% 
  mutate(method = "global")
```

```{r}
egg_near_msq <- egg_near_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "egg") %>% 
  mutate(method = "near gene")

leaf_near_msq <- leaf_near_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "leaf") %>% 
  mutate(method = "near gene")

sperm_near_msq <- sperm_near_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "sperm") %>% 
  mutate(method = "near gene")
```
```{r}
msq_boot <- rbind(
  egg_msq, leaf_msq, sperm_msq,
  egg_near_msq, leaf_near_msq, sperm_near_msq
)
```


```{r}
msq_boot %>% 
  ggplot(aes(x = msq)) +
  facet_grid(sample_type ~ method, scales = "free") +
  geom_histogram(aes(fill = sample_type), alpha = 0.8, bins = 100) +
  geom_point(data = msq.summary, aes(x = msq, y = 20), alpha = 0.8, size = 2.5) + 
  geom_errorbarh(data = msq.summary, aes(xmin = lower.CL, xmax = upper.CL, y = 20), height = 10) +
  labs(x = "mean sum of squares",
       y = "bootstrap distribution") + 
  scale_fill_manual(values = c("tomato1", "darkgreen", "dodgerblue2")) +
  theme_minimal() +
  theme(legend.position = "none", panel.spacing = unit(1, "lines")) +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("gamete_msq.svg", height = 5, width = 5)
```

```{r}
msq.summary %>% 
  ggplot(aes(x = sample_type)) +
  facet_wrap(. ~ method) +
  geom_errorbar(aes(ymin = lower.CL, ymax = upper.CL), width = 0.3) +
  geom_point(aes(y = msq, color =sample_type), alpha = 0.8, size = 2.5) +
  scale_color_manual(values = c("tomato1", "darkgreen", "dodgerblue2")) +
  labs(x = NULL,
       y = "mean sum of squares") + 
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black")) 

# ggsave("gamete_msq.svg", height = 5, width = 5)
```

#leaf z score

```{r}
leaf <- rbind(wt_leaf_s %>% 
                mutate(sample_type = "WT leaf"), 
              ddm1_leaf_s %>% 
                mutate(sample_type = "ddm1ab leaf"),
              drm2_leaf_s %>% 
                mutate(sample_type = "drm2 leaf")) %>% 
  filter(context == "mCHH") %>% 
  group_by(sample_type) %>% 
  mutate(z.score = (mean.methylation  - mean(mean.methylation))/sd(mean.methylation)) %>% 
    ungroup()

head(leaf)
```






```{r}
predicted_values %>% 
ggplot(aes(x = distance, y = predicted_z)) +
  geom_point(data = leaf,
               aes(x = mean.distance,
                   y = z.score,
                   color = sample_type), size = 1, alpha = 0.5) +
  geom_line(aes(group = method, color = method), size = 2, alpha =  0.8, lineend = "round") +
  scale_color_manual(values = c("red", "blue", "darkgreen", "green2","grey40")) + 
  labs(y = "mCHH z-score", 
       x = "distance to nearest gene/bp",
       color = NULL) +
  theme_minimal() +
  theme(legend.position = c(0.6, 0.85)) +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("leaf_regression-z.svg", height = 5, width = 5)
```
#goodness of fit for leaf mutants
```{r}
leaf <- leaf %>% 
  mutate(predicted.z.near = near_coef[2, 1] * mean.distance + near_coef[1, 1]) %>% 
  mutate(predicted.z.global = global_coef[3, 1] * mean.distance^2 + global_coef[2, 1] * mean.distance + global_coef[1, 1]) %>% 
  mutate(xsq.global = (z.score - predicted.z.global)^2 ) %>% 
  mutate(xsq.near = (z.score - predicted.z.near)^2) 

leaf
```
```{r}
obs_wt <- filter(leaf, sample_type == "WT leaf") %>% mutate(df = n() - 1)
obs_ddm <- filter(leaf, sample_type == "ddm1ab leaf") %>% mutate(df = n() - 1)
obs_drm <- filter(leaf, sample_type == "drm2 leaf") %>% mutate(df = n() - 1)
```

```{r}
cor.test(obs_wt$z.score, obs_wt$predicted.z.global)

cor.test(obs_ddm$z.score, obs_ddm$predicted.z.global)

cor.test(obs_drm$z.score, obs_drm$predicted.z.global)
```

```{r}
ddm_ssq <- boot(obs_ddm$xsq.global, R = 1000, statistic = f.sum)
ddm_quantiles <- quantile(ddm_ssq$t, c(alpha/2, 1-alpha/2))

drm_ssq <- boot(obs_drm$xsq.global, R = 1000, statistic = f.sum)
drm_quantiles <- quantile(drm_ssq$t, c(alpha/2, 1-alpha/2))

wt_ssq <- boot(obs_wt$xsq.global, R = 1000, statistic = f.sum)
wt_quantiles <- quantile(wt_ssq$t, c(alpha/2, 1-alpha/2))
```


```{r}
msq.leaf <- c(
  obs_ddm %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747,
  obs_drm %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747,
  obs_wt %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747
)

genotype <- c("ddm1ab", "drm2", "WT")

msq.leaf.global <- data.frame(genotype, msq.leaf) %>% 
  mutate(method = "global") %>% 
  cbind(rbind(ddm_quantiles, drm_quantiles, wt_quantiles))

colnames(msq.leaf.global)[2] <- "msq"
msq.leaf.global
```

```{r}
obs_ddm_near <- filter(leaf, sample_type == "ddm1ab leaf") %>% 
  filter(mean.distance < 750) %>% 
  mutate(df = n() - 1)

obs_drm_near <- filter(leaf, sample_type == "drm2 leaf") %>% 
  filter(mean.distance < 750) %>% 
  mutate(df = n() - 1)

obs_wt_near <- filter(leaf, sample_type == "WT leaf") %>% 
  filter(mean.distance < 750) %>% 
  mutate(df = n() - 1)
```

```{r}
cor.test(obs_wt_near$z.score, obs_wt_near$predicted.z.near)

cor.test(obs_ddm_near$z.score, obs_ddm_near$predicted.z.near)

cor.test(obs_drm_near$z.score, obs_drm_near$predicted.z.near)
```

```{r}
msq_leaf_near <- c(
  obs_ddm_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133,
  obs_drm_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133,
  obs_wt_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133
)

```

```{r}
ddm_near_ssq <- boot(obs_ddm_near$xsq.near, R = 1000, statistic = f.sum_near)
ddm_near_quantiles <- quantile(ddm_near_ssq$t, c(alpha/2, 1-alpha/2))

drm_near_ssq <- boot(obs_drm_near$xsq.near, R = 1000, statistic = f.sum_near)
drm_near_quantiles <- quantile(drm_near_ssq$t, c(alpha/2, 1-alpha/2))

wt_near_ssq <- boot(obs_wt_near$xsq.near, R = 1000, statistic = f.sum_near)
wt_near_quantiles <- quantile(wt_near_ssq$t, c(alpha/2, 1-alpha/2))
```

```{r}
msq.leaf.near <- data.frame(genotype, msq_leaf_near) %>% 
  mutate(method = "near gene") %>% 
  cbind(rbind(ddm_near_quantiles, drm_near_quantiles, wt_near_quantiles))
colnames(msq.leaf.near)[2] <- "msq"

msq.leaf.near
```

```{r}
ddm_msq <- ddm_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(genotype = "ddm1ab") %>% 
  mutate(method = "global")

drm_msq <- drm_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(genotype = "drm2") %>% 
  mutate(method = "global")

wt_msq <- wt_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(genotype = "WT") %>% 
  mutate(method = "global")
```

```{r}
ddm_msq_near <- ddm_near_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(genotype = "ddm1ab") %>% 
  mutate(method = "near gene")

drm_near_msq <- drm_near_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(genotype = "drm2") %>% 
  mutate(method = "near gene")

wt_near_msq <- wt_near_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(genotype = "WT") %>% 
  mutate(method = "near gene")
```

```{r}
msq_boot_leaf <- rbind(
  wt_msq, ddm_msq, drm_msq,
  wt_near_msq, ddm_msq_near, drm_near_msq
)
```




```{r}
msq.leaf.summary <- rbind(msq.leaf.global, msq.leaf.near)
colnames(msq.leaf.summary)[4:5] <- c("lower.CL", "upper.CL")
msq.leaf.summary 
```
```{r}
msq_boot_leaf %>% 
  ggplot(aes(x = msq)) +
  facet_grid(genotype ~ method, scales = "free") +
  geom_histogram(aes(fill = genotype), alpha = 0.8, bins = 100) +
  geom_point(data = msq.leaf.summary, aes(x = msq, y = 20), alpha = 0.8, size = 2.5) + 
  geom_errorbarh(data = msq.leaf.summary, aes(xmin = lower.CL, xmax = upper.CL, y = 20), height = 10) +
  labs(x = "mean sum of squares",
       y = "bootstrap distribution") + 
  scale_fill_manual(values = c("red", "blue", "grey40")) + 
  theme_minimal() +
  theme(legend.position = "none", panel.spacing = unit(1, "lines")) +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("leaf_msq.svg", height = 5, width = 5)
```

```{r}
msq.leaf.summary %>% 
  ggplot(aes(x = genotype)) +
  facet_wrap(. ~ method) +
  geom_errorbar(aes(ymin = lower.CL, ymax = upper.CL), width = 0.3) +
  geom_point(aes(y = msq, color = genotype), alpha = 0.8, size = 2.5) +
  scale_color_manual(values = c("red", "blue", "grey40")) +
  labs(x = NULL,
       y = "mean sum of squares") + 
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black", angle = -90)) +
  theme(axis.text.y = element_text(colour = "black")) 

# ggsave("leaf_msq.svg", height = 5, width = 5)
```
#In endosperm
```{r}
##wt
wt1_en_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt1_en_S191_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt1_en_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt1_en_S191_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt1_en_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt1_en_S191_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt1_en_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/WT1_EN~3.MET", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt2_en_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt2_en_S54_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt2_en_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt2_en_S54_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt2_en_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt2_en_S54_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt2_en_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt2_en_S54_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt3_en_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt3_en_S55_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt3_en_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt3_en_S55_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt3_en_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt3_en_S55_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

wt3_en_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/wt3_en_S55_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-seedling-egg_24siRNAs_intersection.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)
```


```{r}
##drm2 
drm_1_en_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_38en_S194_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_1_en_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_38en_S194_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_1_en_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_38en_S194_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_1_en_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_38E~4.MET", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_2_en_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_39en_S195_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_2_en_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_39en_S195_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_2_en_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_39en_S195_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_2_en_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/529548~1.MET", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_3_en_egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_40en_S196_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_egg-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_3_en_sd <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_40en_S196_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_seedling-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_3_en_sp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_40en_S196_L006_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_sperm-specific_24siRNAs.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

drm_3_en_inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/24nt_siRNA_loci_methylation Feb20/24nt_loci_meth_individual/52_40E~1.MET", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)
```
##cleanup
```{r}
wt1_en_egg1 <- wt1_en_egg %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en1") %>% 
  mutate(bin = "egg-specific") 
colnames(wt1_en_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt1_en_sd1 <- wt1_en_sd %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en1") %>% 
  mutate(bin = "seedling-specific")  
colnames(wt1_en_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt1_en_sp1 <- wt1_en_sp %>% 
  clean() %>%
  mutate(sample_ID = "wt_en1") %>% 
  mutate(bin = "sperm-specific")  
colnames(wt1_en_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt1_en_inter1 <- wt1_en_inter %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en1") %>% 
  mutate(bin = "intersect.")  
colnames(wt1_en_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
wt2_en_egg1 <- wt2_en_egg %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en2") %>% 
  mutate(bin = "egg-specific") 
colnames(wt2_en_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt2_en_sd1 <- wt2_en_sd %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en2") %>% 
  mutate(bin = "seedling-specific")  
colnames(wt2_en_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt2_en_sp1 <- wt2_en_sp %>% 
  clean() %>%
  mutate(sample_ID = "wt_en2") %>% 
  mutate(bin = "sperm-specific")  
colnames(wt2_en_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt2_en_inter1 <- wt2_en_inter %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en2") %>% 
  mutate(bin = "intersect.")  
colnames(wt2_en_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
wt3_en_egg1 <- wt3_en_egg %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en3") %>% 
  mutate(bin = "egg-specific") 
colnames(wt3_en_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt3_en_sd1 <- wt3_en_sd %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en3") %>% 
  mutate(bin = "seedling-specific")  
colnames(wt3_en_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt3_en_sp1 <- wt3_en_sp %>% 
  clean() %>%
  mutate(sample_ID = "wt_en3") %>% 
  mutate(bin = "sperm-specific")  
colnames(wt3_en_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
wt3_en_inter1 <- wt3_en_inter %>% 
  clean() %>% 
  mutate(sample_ID = "wt_en3") %>% 
  mutate(bin = "intersect.")  
colnames(wt3_en_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
drm_1_en_egg1 <- drm_1_en_egg %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en1") %>% 
  mutate(bin = "egg-specific") 
colnames(drm_1_en_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_1_en_sd1 <- drm_1_en_sd %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en1") %>% 
  mutate(bin = "seedling-specific")  
colnames(drm_1_en_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_1_en_sp1 <- drm_1_en_sp %>% 
  clean() %>%
  mutate(sample_ID = "drm_en1") %>% 
  mutate(bin = "sperm-specific")  
colnames(drm_1_en_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_1_en_inter1 <- drm_1_en_inter %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en1") %>% 
  mutate(bin = "intersect.")  
colnames(drm_1_en_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
drm_2_en_egg1 <- drm_2_en_egg %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en2") %>% 
  mutate(bin = "egg-specific") 
colnames(drm_2_en_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_2_en_sd1 <- drm_2_en_sd %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en2") %>% 
  mutate(bin = "seedling-specific")  
colnames(drm_2_en_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_2_en_sp1 <- drm_2_en_sp %>% 
  clean() %>%
  mutate(sample_ID = "drm_en2") %>% 
  mutate(bin = "sperm-specific")  
colnames(drm_2_en_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_2_en_inter1 <- drm_2_en_inter %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en2") %>% 
  mutate(bin = "intersect.")  
colnames(drm_2_en_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
drm_3_en_egg1 <- drm_3_en_egg %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en3") %>% 
  mutate(bin = "egg-specific") 
colnames(drm_3_en_egg1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_3_en_sd1 <- drm_3_en_sd %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en3") %>% 
  mutate(bin = "seedling-specific")  
colnames(drm_3_en_sd1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_3_en_sp1 <- drm_3_en_sp %>% 
  clean() %>%
  mutate(sample_ID = "drm_en3") %>% 
  mutate(bin = "sperm-specific")  
colnames(drm_3_en_sp1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
drm_3_en_inter1 <- drm_3_en_inter %>% 
  clean() %>% 
  mutate(sample_ID = "drm_en3") %>% 
  mutate(bin = "intersect.")  
colnames(drm_3_en_inter1) <- c("chr", "start", "mCG", "mCHG", "mCHH", "sample_ID", "bin")
```

```{r}
endosperm <- rbind(
            wt1_en_egg1, wt1_en_sd1, wt1_en_sp1, wt1_en_inter1,
            wt2_en_egg1, wt2_en_sd1, wt2_en_sp1, wt2_en_inter1,
            wt3_en_egg1, wt3_en_sd1, wt3_en_sp1, wt3_en_inter1,
            drm_1_en_egg1, drm_1_en_sd1, drm_1_en_sp1, drm_1_en_inter1,
            drm_2_en_egg1, drm_2_en_sd1, drm_2_en_sp1, drm_2_en_inter1,
            drm_3_en_egg1, drm_3_en_sd1, drm_3_en_sp1, drm_3_en_inter1
            ) %>% 
  mutate(bins = factor(bin, levels = c(
    "egg-specific",
    "seedling-specific",
    "sperm-specific",
    "intersect."
  ))) %>% 
  mutate(sample_type = "endosperm") %>% 
  mutate(genotype = case_when(
    str_detect(sample_ID, "wt") ~ "WT",
    str_detect(sample_ID, "drm") ~ "drm2"
  )) %>% 
  mutate(genotype = factor(genotype, levels = c("drm2", "WT"))) %>%  
   gather("context", "methylation", 3:5) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(context, sample_type, genotype, chr, start, bins) %>% 
  summarise(methylation = mean(methylation)) %>% 
  ungroup()
```

```{r}
endosperm_s <- endosperm %>%
  full_join(distance_to_gene, by = c("chr", "start", "bins" ="bin")) %>% 
  filter(distance < quantile(distance, 0.95)) %>%
  filter(distance > 100)  %>% 
  mutate(bucket = ntile(distance, 750)) %>% 
  filter(is.na(methylation) == F) %>% 
  group_by(bucket, context, genotype, sample_type) %>%
  summarise(mean.distance = mean(distance), mean.methylation = mean(methylation)) %>% 
  ungroup()


head(endosperm_s, 10)
```
```{r}
leaf_endo <- wt_leaf_s %>% 
  mutate(sample_type = "leaf") %>% 
  mutate(genotype = "WT") %>% 
  select(bucket, context, genotype, sample_type, mean.distance, mean.methylation) %>% 
  rbind(endosperm_s) %>% 
  filter(context == "mCHH") %>% 
  group_by(sample_type, genotype) %>% 
  mutate(z.score = (mean.methylation - mean(mean.methylation))/sd(mean.methylation)) %>% 
  ungroup()


head(leaf_endo)
```
```{r}
predicted_values %>% 
ggplot(aes(x = distance, y = predicted_z)) +
  geom_point(data = leaf_endo,
               aes(x = mean.distance,
                   y = z.score,
                   color = paste(genotype, sample_type)), size = 1, alpha = 0.5) +
  geom_line(aes(group = method, color = method), size = 2, alpha =  0.8, lineend = "round") +
  scale_color_manual(values = c("blue", "darkgreen", "green2", "gold4", "black")) + 
  labs(y = "mCHH z-score", 
       x = "distance to nearest gene/bp",
       color = NULL) +
  theme_minimal() +
  theme(legend.position = c(0.6, 0.85)) +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("leaf_vs_endosperm_regression_dots.svg", height = 5, width = 5)
```
#goodness of fit for endosperm
```{r}
leaf_endo <- leaf_endo %>% 
  mutate(predicted.z.near = near_coef[2, 1] * mean.distance + near_coef[1, 1]) %>% 
  mutate(predicted.z.global = global_coef[3, 1] * mean.distance^2 + global_coef[2, 1] * mean.distance + global_coef[1, 1]) %>% 
  mutate(xsq.global = (z.score - predicted.z.global)^2 ) %>% 
  mutate(xsq.near = (z.score - predicted.z.near)^2) 

leaf_endo
```
```{r}
obs_leaf <- filter(leaf_endo, sample_type == "leaf", genotype == "WT") %>% mutate(df = n() - 1)
obs_drm_en <- filter(leaf_endo, genotype == "drm2", sample_type == "endosperm") %>% mutate(df = n() - 1)
obs_wt_en <- filter(leaf_endo, sample_type == "endosperm", genotype == "WT") %>% mutate(df = n() - 1)
```

```{r}
cor.test(obs_leaf$z.score, obs_leaf$predicted.z.global)
cor.test(obs_drm_en$z.score, obs_drm_en$predicted.z.global)
cor.test(obs_wt_en$z.score, obs_wt_en$predicted.z.global)
```

```{r}
leaf_ssq <- boot(obs_leaf$xsq.global, R = 1000, statistic = f.sum)
leaf_quantiles <- quantile(leaf_ssq$t, c(alpha/2, 1-alpha/2))

drm_en_ssq <- boot(obs_drm_en$xsq.global, R = 1000, statistic = f.sum)
drm_en_quantiles <- quantile(drm_en_ssq$t, c(alpha/2, 1-alpha/2))

wt_en_ssq <- boot(obs_wt_en$xsq.global, R = 1000, statistic = f.sum)
wt_en_quantiles <- quantile(wt_en_ssq$t, c(alpha/2, 1-alpha/2))
```

```{r}
msq.endo <- c(
  obs_leaf %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747,
  obs_drm_en %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747,
  obs_wt_en %>% summarise(ssq = sum(xsq.global)) %>% as.numeric()/747
)

sample_type <- c("WT leaf", "drm2 endosperm", "WT endosperm")


msq.endo.global <- data.frame(sample_type, msq.endo) %>% 
  mutate(method = "global") %>% 
  cbind(rbind(leaf_quantiles, drm_en_quantiles, wt_en_quantiles))

colnames(msq.endo.global)[2] <- "msq"
msq.endo.global
```
```{r}
obs_leaf_near <- filter(leaf_endo, sample_type == "leaf", genotype == "WT") %>% 
  filter(mean.distance < 750)

obs_drm_en_near <- filter(leaf_endo, genotype == "drm2", sample_type == "endosperm") %>% 
  filter(mean.distance < 750)


obs_wt_en_near <- filter(leaf_endo, sample_type == "endosperm", genotype == "WT") %>% 
  filter(mean.distance < 750)
```

```{r}
cor.test(obs_leaf_near$z.score, obs_leaf_near$predicted.z.near)

cor.test(obs_drm_en_near$z.score, obs_drm_en_near$predicted.z.near)

cor.test(obs_wt_en_near$z.score, obs_wt_en_near$predicted.z.near)
```


```{r}
msq.endo.near <- c(
  obs_leaf_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133,
  obs_drm_en_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133,
  obs_wt_en_near %>% summarise(ssq = sum(xsq.near)) %>% as.numeric()/133
)
```

```{r}
leaf_ssq_near <- boot(obs_leaf_near$xsq.near, R = 1000, statistic = f.sum_near)
leaf_quantiles_near <- quantile(leaf_ssq_near$t, c(alpha/2, 1-alpha/2))

drm_en_ssq_near <- boot(obs_drm_en_near$xsq.near, R = 1000, statistic = f.sum_near)
drm_en_quantiles_near <- quantile(drm_en_ssq_near$t, c(alpha/2, 1-alpha/2))

wt_en_ssq_near <- boot(obs_wt_en_near$xsq.near, R = 1000, statistic = f.sum_near)
wt_en_quantiles_near <- quantile(wt_en_ssq_near$t, c(alpha/2, 1-alpha/2))
```

```{r}
msq.endo.near <- data.frame(sample_type, msq.endo.near) %>% 
  mutate(method = "near gene") %>% 
  cbind(rbind(leaf_quantiles_near, drm_en_quantiles_near, wt_en_quantiles_near))

colnames(msq.endo.near)[2] <- "msq"
msq.endo.near
```
```{r}
msq.endo.summary <- rbind(msq.endo.global, msq.endo.near
                         ) %>% 
  separate(sample_type, c("genotype", "sample_type"), sep = " ")
colnames(msq.endo.summary)[5:6] <- c("lower.CL", "upper.CL")

msq.endo.summary <- msq.endo.summary %>% 
  rbind( msq.leaf.summary %>% 
                            filter(genotype == "drm2") %>%
                            mutate(sample_type = "leaf") %>% 
                            select(genotype, sample_type, msq, method, lower.CL, upper.CL))
```


```{r}
drm_en_msq <- drm_en_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "endosperm") %>%
  mutate(genotype = "drm2") %>% 
  mutate(method = "global")

wt_en_msq <- wt_en_ssq$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "endosperm") %>%
  mutate(genotype = "WT") %>% 
  mutate(method = "global")

leaf_msq <- leaf_msq %>% 
  mutate(genotype = "WT") %>% 
  select(msq, sample_type, genotype, method)

drm_msq <- drm_msq %>% 
  mutate(genotype = "drm2") %>%
  mutate(sample_type = "leaf") %>%
  select(msq, sample_type, genotype, method)
```

```{r}
drm_en_msq_near <- drm_en_ssq_near$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "endosperm") %>%
  mutate(genotype = "drm2") %>% 
  mutate(method = "near gene")

wt_en_msq_near <- wt_en_ssq_near$t %>% 
  as.data.frame() %>% 
  mutate(msq = V1) %>% 
  select(-V1) %>% 
  mutate(sample_type = "endosperm") %>%
  mutate(genotype = "WT") %>% 
  mutate(method = "near gene")

leaf_msq_near <- leaf_near_msq %>% 
  mutate(genotype = "WT") %>% 
  select(msq, sample_type, genotype, method)

drm_near_msq <- drm_near_msq %>% 
  mutate(genotype = "drm2") %>%
  mutate(sample_type = "leaf") %>%
  select(msq, sample_type, genotype, method)
```

```{r}
msq_boot_endo <- rbind(
  drm_en_msq, wt_en_msq, leaf_msq, drm_msq,
  drm_en_msq_near, wt_en_msq_near, leaf_msq_near, drm_near_msq
)

tail(msq_boot_endo)
```

```{r}
msq_boot_endo %>% 
  mutate(genotype.f = factor(genotype, levels = c(
    "WT", "drm2"
  ))) %>% 
  ggplot(aes(x = msq)) +
  facet_grid(genotype ~ method, scales = "free") +
  geom_histogram(aes(fill = sample_type), alpha = 0.8, bins = 100) +
  geom_point(data = msq.endo.summary, aes(x = msq, y = 20), alpha = 0.8, size = 2.5) + 
  geom_errorbarh(data = msq.endo.summary, aes(xmin = lower.CL, xmax = upper.CL, y = 20), height = 10) +
  labs(x = "mean sum of squares",
       y = "bootstrap distribution",
       fill = NULL) + 
  scale_fill_manual(values = c("gold4", "darkgreen")) +
  theme_minimal() +
  theme(legend.position = "bottom", panel.spacing = unit(1, "lines")) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black"))

ggsave("endosperm_msq.svg", height = 5, width = 5)
```


```{r}
msq.endo.summary %>% 
  ggplot(aes(x = sample_type)) +
  facet_grid(. ~ method + genotype, scales = "free") +
  geom_errorbar(aes(ymin = lower.CL, ymax = upper.CL), width = 0.3) +
  geom_point(aes(y = msq, color =sample_type), alpha = 0.8, size = 2.5) +
  # scale_color_manual(values = c("blue", "gold4", "darkgreen")) +
  scale_color_manual(values = c("gold4", "darkgreen")) +
  labs(x = NULL,
       y = "mean sum of squares") + 
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y = element_text(colour = "black"))

# ggsave("endosperm_msq.svg", height = 5, width = 5)
```


Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
