---
title: "miR"
author: "Chenxin Li"
date: "February 2, 2019"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(vegan)
library(edgeR)
library(dynamicTreeCut)
library(RColorBrewer)
library(stringr)
library(svglite)
```
#load data


```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/sample_des.xlsx") 
sample_des <- sample_des %>% 
  arrange(sample_ID) 

#sample_des 

total_siRNA <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS08/TE/total_siRNA.xlsx")  

sample_des_lib <- sample_des %>% 
  select(-library)%>% 
  inner_join(total_siRNA, by = "sample_ID") 

sample_des_lib 
```

```{r}
B15 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/Bulk-15_S26_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

B20 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/Bulk-20_S27_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

B25 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/Bulk-25_S28_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)


BS1 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/blender-sperm-1_S30_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BS2 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderSperm2_S8_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BS4a <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderSperm4a_S173_L007_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BS4b <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderSperm4b_S176_L007_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BS6a <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderSperm6a_S174_L007_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BS6b <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderSperm6b_S177_L007_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BV1 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderVegetativeCell1_S4_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

BV2 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/BlenderVegetativeCell2_S6_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

ddm1 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/ddm1ab.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

ddm1_con <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/ddm1ab_control.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

drm2 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/drm2-T-DNA.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

drm2_con <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/drm2-T-DNA_control.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

E2 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/eggcell2_combined.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

E3 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/eggcell3_combined.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

E4 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/eggcell4_combined.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

E9a <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/egg-9a_S25_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

E9b <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/egg-9b_S26_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

E9c <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/egg-9c_S27_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

OV1a <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/ovary-no-egg-1a_S31_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

OV1b <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/ovary-no-egg-1b_S32_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

OV2 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/Ovarynoegg2_S5_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

P18 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/protoplast-18_S29_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

P20 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/protoplast-20_S30_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

P25 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/protoplast-25_S31_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

root <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/SRR5713891.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

SD4 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/seedling4_S3_L004_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

SP5b <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/sperm-5b_S28_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

SP5c <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/miR_coverage/miR_coverage/sperm-5c_S29_L002_R1_001.fastq.gz.trimmed.fastq.gz.miR.cov", 
     "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)
```
#data arrangement
```{r}
cts <- B15[, 9:10] %>% 
  full_join(B20[, 9:10], by = "X9") %>% 
  full_join(B25[, 9:10], by = "X9") %>% 
  full_join(BS1[, 9:10], by = "X9") %>% 
  full_join(BS2[, 9:10], by = "X9") %>% 
  full_join(BS4a[, 9:10], by = "X9") %>% 
  full_join(BS4b[, 9:10], by = "X9") %>% 
  full_join(BS6a[, 9:10], by = "X9") %>% 
  full_join(BS6b[, 9:10], by = "X9") %>% 
  full_join(BV1[, 9:10], by = "X9") %>% 
  full_join(BV2[, 9:10], by = "X9") %>% 
  full_join(ddm1[, 9:10], by = "X9") %>% 
  full_join(ddm1_con[, 9:10], by = "X9") %>% 
  full_join(drm2[, 9:10], by = "X9") %>% 
  full_join(drm2_con[, 9:10], by = "X9") %>% 
  full_join(E2[, 9:10], by = "X9") %>% 
  full_join(E3[, 9:10], by = "X9") %>% 
  full_join(E4[, 9:10], by = "X9") %>% 
  full_join(E9a[, 9:10], by = "X9") %>% 
  full_join(E9b[, 9:10], by = "X9") %>% 
  full_join(E9c[, 9:10], by = "X9") %>% 
  full_join(OV1a[, 9:10], by = "X9") %>% 
  full_join(OV1b[, 9:10], by = "X9") %>% 
  full_join(OV2[, 9:10], by = "X9") %>% 
  full_join(P18[, 9:10], by = "X9") %>% 
  full_join(P20[, 9:10], by = "X9") %>% 
  full_join(P25[, 9:10], by = "X9") %>% 
  full_join(root[, 9:10], by = "X9") %>% 
  full_join(SD4[, 9:10], by = "X9") %>% 
  full_join(SP5b[, 9:10], by = "X9") %>% 
  full_join(SP5c[, 9:10], by = "X9") %>% 
  separate(X9, c("A", "B", "C", "D"), sep = ";", remove = T) %>% 
  select(-A, -B, -D) %>% 
  separate(C, c("C1", "miR"), remove = T, sep = "=") %>% 
  select(-C1)
  
colnames(cts)[2:32] <- sample_des$sample_ID 

head(cts) 
write_excel_csv(cts, "miR_cts.csv")

```
```{r}
osa_mirBase22 <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/osa_mirBase22.txt", header=FALSE) 

osa_mir <- osa_mirBase22[, 1:2]
colnames(osa_mir) <- c("tag", "sequence") 
osa_mir <- osa_mir%>% 
  separate(tag, c("T1", "T2", "T3", "T4", "T5"), sep = " ", remove = T) %>% 
  mutate(miR = paste("osa", T5, sep = "-")) %>% 
  select(-T1, -T3, -T4, -T5) %>% 
  arrange(T2)

osa_mir <- osa_mir[osa_mir$miR %in% cts$miR, ]
head(osa_mir, 10)
```
```{r}
osa_mir_unique <- distinct(osa_mir, sequence, .keep_all = T)
osa_mir_dup <- anti_join(osa_mir, osa_mir_unique, by = c("miR", "sequence"))
dim(osa_mir_unique)

head(osa_mir_unique, 10)

write.csv(osa_mir, "osa_mir_seq.csv")
write.csv(osa_mir_dup, "osa_mir_seq_duplicated.csv")
```

```{r}
cts_seq <- cts %>% 
  full_join(osa_mir, by = "miR") %>% 
  #mutate(n.ch = nchar(miR)) %>% 
  #mutate(fam = substr(miR, 1, (n.ch - 1))) %>% 
  gather("sample_ID", "counts", 2:32) %>% 
  group_by(sequence, sample_ID) %>%
  summarise(sum.count = sum(counts)) %>% 
  spread(sample_ID, sum.count) %>% 
  full_join(osa_mir_unique, by = "sequence") %>% 
  arrange(T2) %>% 
  select(-T2)

cts_seq <- cts_seq[complete.cases(cts_seq), ]
cts_seq %>% head(10)  


dim(cts_seq)[1] == dim(osa_mir_unique)[1]

#cts_seq %>% filter(str_detect(miR, "159"))
```

#EdgeR
```{r}
cts_matrix <- cts[, 2:32] %>% as.data.frame()
row.names(cts_matrix) <- cts$miR

y <- DGEList(counts=cts_matrix ,group=sample_des$sample_type, lib.size = sample_des_lib$total_smRNA)    
y$samples

keep <- rowSums(cpm(y)> 1) >= 3 
y <- y[keep, , keep.lib.sizes = T]
y <- calcNormFactors(y, method = "TMM")
#calcNormFactors calculates normalization factors using TMM
y$samples

sample_des$class <- paste(sample_des$sample_type, sample_des$genotype, sep = ":")
design <- model.matrix(~0 + sample_des$sample_type)
y <- estimateDisp(y, design)
#plotBCV(y)
```
#PCA plots
```{r}
cpm <- cpm(y, log =F) %>% as.data.frame()
logcpm <- cpm(y, log =T) %>% as.data.frame()    #lib.size = sample_des$total_smRNA
write.csv(cpm, "gene_siRNA_CPM.csv")
#adonis((t(log(cpm+1))) ~ sample_type * genotype, sample_des, method= "e")
adonis((t(logcpm)) ~ sample_type * genotype, sample_des, method= "e")
```
```{r}
pal <- c("tomato1", "darkgreen", "orangered3", "grey40", "seagreen", "dodgerblue2") 
```

```{r}
pc <- prcomp(t(logcpm))
pc_importance <- as.data.frame(t(summary(pc)$importance))
pc_importance

ggplot(pc_importance, aes(x = 1:length( `Proportion of Variance`), y = `Proportion of Variance`) ) +
  geom_bar(stat = 'identity') +
  xlab('dimension') +
  theme_bw()

ggplot(pc_importance, aes(x = 1:length(`Cumulative Proportion`), y = `Cumulative Proportion`) ) +
  geom_point() +
  xlab('dimension') +
  ylab('cumulative variance') +
  ylim(c(0, 1.1))+
  theme_bw()

pc_nice <- cbind(sample_des, pc$x)
pc_nice <- pc_nice %>% 
  mutate(type = case_when(
    sample_type == "seedling shoot" |
      sample_type == "sd protoplast" |
      sample_type == "root" |
      sample_type == "leaf" ~ "vegetative tissue",
    sample_type == "egg" |
      sample_type == "ovary" ~ "female",
    sample_type == "sperm" |
      sample_type == "pollen VC" ~ "male"
  )) %>% 
  mutate(type = factor(type, levels = c("female", "vegetative tissue", "male")))


```
```{r}
pc_nice %>% 
  filter(sample_type != "sd protoplast") %>% 
  filter(sample_type != "pollen VC") %>% 
ggplot(aes(PC1, PC2)) +
  stat_ellipse(aes(fill = type), alpha = 0.3, level = 0.9, type = "t", geom = "polygon") +
  geom_point(aes(color = sample_type), size = 3.5, alpha = 0.8) +
  labs(x = paste("PC1 (", pc_importance[1, 2]*100, "% of Variance)", sep = ""), 
       y = paste("PC2 (", pc_importance[2, 2]*100, "% of Variance)", sep = ""),
       fill = NULL,
       color = NULL) +
  scale_color_manual(values = pal) +
  #scale_shape_manual(values = c(15, 17, 16)) +
  #scale_fill_manual() + 
  theme_minimal()+
  theme(legend.position = "right")+
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black"))

ggsave(filename = "miR_PCA.svg", width = 6, height = 5)
```

##Anova on PC plot coordinate
```{r}
model_x <- lm(PC1 ~ sample_type, pc_nice)
anova(model_x)

est_x <- emmeans(model_x, pairwise ~ sample_type)
est_x_bar <- as.data.frame(summary(est_x)$emmeans)
model_y <- lm(PC2 ~ sample_type, pc_nice)
anova(model_y)

est_y <- emmeans(model_y, pairwise ~ sample_type)
est_y_bar <- as.data.frame(summary(est_y)$emmeans)
cld.x <- CLD(est_x$emmeans, Letters = letters) %>% as.data.frame()
cld.y <- CLD(est_y$emmeans, Letters = letters) %>% as.data.frame()
```
```{r}
colnames(cld.x)[2] <- "PC1"
colnames(cld.y)[2] <- "PC2"

cld.xy <- cld.x %>% 
  full_join(cld.y, by = "sample_type")

head(cld.xy)
```

```{r}
# ggplot(cld.xy, aes(PC1, PC2)) +
#   stat_ellipse(data = pc_nice, aes(fill = type), alpha = 0.3, level = 0.9, type = "t", geom = "polygon") +
#   geom_errorbar(aes(ymin = lower.CL.y, ymax = upper.CL.y, color = sample_type), size = 1, alpha = 0.9) +
#   geom_errorbarh(aes(xmin = lower.CL.x, xmax = upper.CL.x, color = sample_type), size = 1, alpha = 0.9) +
#   #geom_point(data = pc_nice, aes(x = PC1, y = PC2, color = sample_type)) +
#   geom_point(aes(color = sample_type), size = 3.5, alpha = 0.8) +
#   labs(x = paste("PC1 (", pc_importance[1, 2]*100, "% of Variance)", sep = ""), 
#        y = paste("PC2 (", pc_importance[2, 2]*100, "% of Variance)", sep = ""),
#        color = NULL,
#        fill = NULL) +
#   scale_color_manual(values = pal)  + 
#   theme_minimal()+
#   theme(text = element_text(size= 14, face="bold")) +
#   theme(axis.text.x=element_text(colour = "black")) +
#   theme(axis.text.y=element_text(colour = "black"))
# 
# ggsave(filename = "miR_PCA_errorbar.png")
```
##coordinate p value matrix
```{r}
self <- cbind(
    c("egg cell", "leaf", "ovary no egg", "pollen vegetative cell",
                      "root", "seedling protoplast", "seedling segment", "sperm cell"),
    c("egg cell", "leaf", "ovary no egg", "pollen vegetative cell",
                      "root", "seedling protoplast", "seedling segment", "sperm cell"),
    c(rep(1, 8))
    ) %>% 
  as.data.frame() 

colnames(self) <- c("sample_type1", "sample_type2", "p.value")
self$p.value <- as.numeric(self$p.value)
#self
```

```{r}
est_x1 <- est_x$contrasts %>%
  as.data.frame() %>% 
  separate(contrast, c("sample_type1", "sample_type2"), remove = T, sep = " - ") %>% 
  select(sample_type1, sample_type2, p.value) %>% 
  rbind(self) %>% 
  spread("sample_type2", "p.value") 

est_x_mat <- est_x1[, -1] 
row.names(est_x_mat) <- est_x1$sample_type1

#est_x_mat
```
```{r}
est_y1 <- est_y$contrasts %>%
  as.data.frame() %>% 
  separate(contrast, c("sample_type1", "sample_type2"), remove = T, sep = " - ") %>% 
  select(sample_type1, sample_type2, p.value) %>%
  rbind(self) %>% 
  spread("sample_type2", "p.value") 

est_y_mat <- est_y1[, -1] %>% as.matrix()
row.names(est_y_mat) <- est_y1$sample_type1

#est_y_mat %>% t()
```
```{r}
p_mat <- matrix(NA, nrow = 8, ncol = 8)
p_mat[upper.tri(p_mat)] <- est_x_mat[upper.tri(est_x_mat)]
p_mat[lower.tri(p_mat)] <- est_y_mat[upper.tri(est_y_mat)]

diag(p_mat) <- 1
colnames(p_mat) <- self$sample_type2

p_mat <- p_mat %>% 
  as.data.frame() %>% 
  mutate(sample_type1 = self$sample_type1) %>% 
  gather("sample_type2", "p.value", 1:8)
```

```{r}
p_mat %>% 
  ggplot(aes(x = sample_type1, y = sample_type2)) +
  geom_tile(aes(fill = p.value)) +
  geom_text(aes(label = p.value %>% round(2)), size = 5, fontface = "bold") +
  scale_fill_gradientn(colours = rev(brewer.pal(11, "Spectral"))) +
  scale_x_discrete(labels = c("egg", "leaf", "ovary", "pollen VC", 
                             "root", "sd protoplast", "sd segment", "sperm")) +
  scale_y_discrete(labels = c("egg", "leaf", "ovary", "pollen VC", 
                           "root", "sd protoplast", "sd segment", "sperm")) +
  labs(x = NULL, 
       y = NULL,
       subtitle = "upper tri. for x coordinates; lower tri. for y coordinates") +
  theme_minimal() +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black"))

ggsave(filename = "p_heatmap.png")
```

#correlation heatmaps
```{r}
cpm_table <- cpm %>% 
  mutate(miR = row.names(cpm)) %>% 
  as.data.frame()

head(cpm_table)
```
```{r}
cor_matrix <- cor(cpm %>% 
                    select(-ddm1, -drm2))
sample_des_wt <- filter(sample_des, genotype == "WT")

#cor_matrix
cor_tidy <- cor_matrix %>% 
  as.data.frame() %>% 
  mutate(sample_ID = row.names(cor_matrix)) %>% 
  gather("sample_ID.y", "cor", 1:29) %>% 
  full_join(sample_des_wt, by ="sample_ID") %>% 
  full_join(sample_des_wt %>% 
               mutate(sample_ID.y = sample_ID), by = "sample_ID.y") %>% 
  select(sample_ID.x, sample_ID.y, cor, sample_type.x, sample_type.y) %>% 
  group_by(sample_type.x, sample_type.y) %>% 
  summarise(mean.cor = mean(cor)) %>% 
  spread(sample_type.y, mean.cor)
cor_tidy1 <- cor_tidy[, -1] %>% 
  as.data.frame()
row.names(cor_tidy1) <- cor_tidy$sample_type.x
cor_tidy1[upper.tri(cor_tidy1)] <- NA
cor_tidy1
```
```{r}
cor_tidy <- cor_tidy1 %>% 
  as.data.frame() %>% 
  mutate(sample_type.x = row.names(cor_tidy1)) %>% 
  gather("sample_type.y", "mean.cor", 1:8) %>% 
  mutate(cor1 = case_when(
    mean.cor == 1 ~ paste(mean.cor %>% round(3), "*", sep = ""),
    mean.cor != 1 ~ paste(mean.cor %>% round(3), "", sep = "")
  )) 
cor_tidy
```
```{r}
cor_tidy %>% 
  ggplot(aes(x = sample_type.x, y = sample_type.y)) +
  geom_tile(aes(fill = mean.cor)) +
  scale_fill_gradientn(colours = brewer.pal(9, "YlOrRd")[c(1:9)], na.value = NA) + 
  geom_text(aes(label = cor1), size = 5, fontface = "bold") +
  scale_x_discrete(labels = c("egg", "leaf", "ovary", "pollen VC", 
                             "root", "sd protoplast", "seedling", "sperm")) +
  scale_y_discrete(labels = c("egg", "leaf", "ovary", "pollen VC", 
                           "root", "sd protoplast", "seedling", "sperm")) +
  labs(x = NULL,
       y = NULL,
       fill = "cor") +
  ggtitle("correlation heatmap for miRNA") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))


ggsave(filename = "miR_cor_heatmap.png")
```
#miRNA clusters
egg vs. seedling vs sperm 
```{r}
logcpm_l <- logcpm %>% 
  mutate(miR = row.names(logcpm)) %>% 
  gather("sample_ID", "logCPM", 1:31) %>% 
  group_by(miR) %>% 
  mutate(z.s = (logCPM - mean(logCPM))/sd(logCPM)) %>%
  mutate(sample_type = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4" |
      sample_ID == "E9a" |
      sample_ID == "E9b" |
      sample_ID == "E9c" ~ "egg",
    sample_ID == "SP5b" |
      sample_ID == "SP5c" |
      sample_ID == "BS1" |
      sample_ID == "BS2" |
      sample_ID == "BS4a" |
      sample_ID == "BS4b" |
      sample_ID == "BS6a" |
      sample_ID == "BS6b"  ~ "sperm", 
    sample_ID == "BV1" |
      sample_ID == "BV2" ~ "pollen VC", 
    sample_ID == "B15" |
      sample_ID == "B20" |
      sample_ID == "B25" |
      sample_ID == "SD4" ~ "seedling shoot", 
    sample_ID == "P18" |
      sample_ID == "P20" |
      sample_ID == "P25" ~ "sd protoplast", 
    sample_ID == "OV1a" |
      sample_ID == "OV1b" |
      sample_ID == "OV2" ~ "ovary",
    sample_ID == "root" ~ "root",
    sample_ID == "ddm1" |
      sample_ID == "ddm1_con" |
      sample_ID == "drm2" |
      sample_ID == "drm2_con" ~ "leaf"
  )) %>% 
  filter(sample_type == "egg"|
           sample_type == "seedling shoot" |
           sample_type == "sperm") %>%   
  mutate(type = case_when(
    sample_type == "egg" |
      sample_type == "ovary" ~ "female",
    sample_type == "sperm" |
      sample_type == "pollen VC" ~ "male",
    sample_type == "seedling shoot" | 
      sample_type == "sd protoplast" |
      sample_type == "leaf" |
      sample_type == "root" ~ "vegetative tissues"
  )) %>% 
  group_by(type, miR) %>% 
  summarise(Mean = mean(z.s),
            logCPM = mean(logCPM)) %>% 
  as.data.frame()
  
colnames(logcpm_l) <- c("type", "miR", "z.score", "logCPM")
logcpm_l %>% head(10)
```
```{r}
z <- logcpm_l[,1:3] %>% 
  spread(type, z.score) %>% 
  as.data.frame()
row.names(z) <- z$miR
z
```
```{r}
all.dist <- dist(z[2:4], method = "e") 
all.hc <- hclust(all.dist, method = "complete")
plot(all.hc)
```
```{r}
all.clust.obj <- cutreeHybrid(dendro = all.hc, distM = as.matrix(all.dist), minClusterSize = 5)

temp2 <- all.clust.obj$labels %>% 
  as.data.frame()
colnames(temp2) <- "cluster_name"
temp3 <- cbind(colnames(as.matrix(all.dist)), temp2) %>% 
  as.data.frame()
colnames(temp3)[1] <- "miR"

temp3

unique(temp3$cluster_name) %>% length()

logcpm_clust <- temp3 %>% 
  inner_join(logcpm_l, by = "miR")

logcpm_clust$type <- factor(logcpm_clust$type, levels = c("female", "vegetative tissues", "male"))
logcpm_clust
```
```{r}
write_excel_csv(temp3, "cluster_assignment.csv")
temp3 %>% 
  group_by(cluster_name) %>% 
  count() %>% 
  ggplot(aes(x = cluster_name, y = n)) +
  geom_point() +
  labs(x = "cluster",
       y = "number of miR") +
  theme_bw() 
```
```{r}
ggplot(logcpm_clust,
       aes(x = type, y = z.score)) +
  facet_wrap(~ cluster_name, ncol = 6)+
  #geom_point(size = 1, alpha = 0.5, aes(color = type)) +
  geom_line(aes(group = miR), alpha = 0.8, color = "grey40") +
  stat_summary(geom = "line", aes(group = cluster_name, color = as.factor(cluster_name)), fun.y = mean, size = 2, alpha = 0.8) +
  stat_summary(geom = "linerange", aes(color = as.factor(cluster_name)), fun.data = "mean_cl_boot") +
  scale_x_discrete(labels = c("egg", "seedling", "sperm")) +
  labs(x = NULL) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size = 17, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black"))

ggsave(filename = "miR_clusters.svg", width = 6, height = 6)
```
```{r}
ggplot(logcpm_clust,
       aes(x = type, y = miR)) +
  facet_grid(cluster_name ~ ., scales = "free") + 
  geom_tile(aes(fill = z.score)) +
  scale_fill_gradientn(colours = rev(brewer.pal(11, "Spectral"))) +
  labs(fill = "z score",
       x = NULL) +
  scale_x_discrete(labels = c("egg", "seedling", "sperm")) +
  theme_minimal() +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  #theme(axis.text.y=element_text(colour = "black")) +
  theme(axis.text.y = element_blank()) 

ggsave(filename = "miR_heatmap.png", width = 5, height = 5, units = "in")
```
#miR pseudo PCA based on miR clusters

```{r}

logcpm_l1 <- logcpm %>% 
  mutate(miR = row.names(logcpm)) %>% 
  gather("sample_ID", "logCPM", 1:31) %>% 
  group_by(miR) %>% 
  mutate(z.s = (logCPM - mean(logCPM))/sd(logCPM)) %>%
  mutate(sample_type = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4" |
      sample_ID == "E9a" |
      sample_ID == "E9b" |
      sample_ID == "E9c" ~ "egg",
    sample_ID == "SP5b" |
      sample_ID == "SP5c" |
      sample_ID == "BS1" |
      sample_ID == "BS2" |
      sample_ID == "BS4a" |
      sample_ID == "BS4b" |
      sample_ID == "BS6a" |
      sample_ID == "BS6b"  ~ "sperm", 
    sample_ID == "BV1" |
      sample_ID == "BV2" ~ "pollen VC", 
    sample_ID == "B15" |
      sample_ID == "B20" |
      sample_ID == "B25" |
      sample_ID == "SD4" ~ "seedling shoot", 
    sample_ID == "P18" |
      sample_ID == "P20" |
      sample_ID == "P25" ~ "sd protoplast", 
    sample_ID == "OV1a" |
      sample_ID == "OV1b" |
      sample_ID == "OV2" ~ "ovary",
    sample_ID == "root" ~ "root",
    sample_ID == "ddm1" |
      sample_ID == "ddm1_con" |
      sample_ID == "drm2" |
      sample_ID == "drm2_con" ~ "leaf"
  )) %>% 
   filter(sample_type == "egg"|
           sample_type == "seedling shoot" |
           sample_type == "sperm") %>%
  mutate(type = case_when(
    sample_type == "egg" |
      sample_type == "ovary" ~ "female",
    sample_type == "sperm" |
      sample_type == "pollen VC" ~ "male",
    sample_type == "seedling shoot" | 
      sample_type == "sd protoplast" |
      sample_type == "leaf" |
      sample_type == "root" ~ "vegetative tissues"
  )) %>% 
  as.data.frame() 
  
#logcpm_l1 %>% head(10)

clust_log <- temp3 %>% 
  inner_join(logcpm_l1, by = "miR") %>% 
  mutate(type = factor(type, levels = c("female", "vegetative tissues", "male")))
```
```{r}
sex <- clust_log %>% 
  group_by(cluster_name, sample_ID) %>% 
  summarise(mean.logCPM = mean(logCPM)) %>% 
  mutate(cluster = paste("cluster", cluster_name, sep = "_")) %>% 
  ungroup() %>% 
  select(-cluster_name) %>% 
  spread(cluster, mean.logCPM) %>% 
  mutate(sample_type = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4" |
      sample_ID == "E9a" |
      sample_ID == "E9b" |
      sample_ID == "E9c" ~ "egg",
    sample_ID == "SP5b" |
      sample_ID == "SP5c" |
      sample_ID == "BS1" |
      sample_ID == "BS2" |
      sample_ID == "BS4a" |
      sample_ID == "BS4b" |
      sample_ID == "BS6a" |
      sample_ID == "BS6b"  ~ "sperm", 
    sample_ID == "BV1" |
      sample_ID == "BV2" ~ "pollen VC", 
    sample_ID == "B15" |
      sample_ID == "B20" |
      sample_ID == "B25" |
      sample_ID == "SD4" ~ "seedling shoot", 
    sample_ID == "P18" |
      sample_ID == "P20" |
      sample_ID == "P25" ~ "sd protoplast", 
    sample_ID == "OV1a" |
      sample_ID == "OV1b" |
      sample_ID == "OV2" ~ "ovary",
    sample_ID == "root" ~ "root",
    sample_ID == "ddm1" |
      sample_ID == "ddm1_con" |
      sample_ID == "drm2" |
      sample_ID == "drm2_con" ~ "leaf"
  )) %>% 
   filter(sample_type == "egg"|
           sample_type == "seedling shoot" |
           sample_type == "sperm") %>%
  mutate(type = case_when(
    sample_type == "egg" |
      sample_type == "ovary" ~ "female",
    sample_type == "sperm" |
      sample_type == "pollen VC" ~ "male",
    sample_type == "seedling shoot" | 
      sample_type == "sd protoplast" |
      sample_type == "leaf" |
      sample_type == "root" ~ "vegetative tissues"
  )) %>% 
   mutate(type = factor(type, levels = c("female", "vegetative tissues", "male")))

sex %>% head(10)
```

```{r}
pseudo_pc <- pc_nice %>% 
  select(sample_ID, sample_type, PC1, PC2, type) %>% 
  inner_join(sex, by = c("sample_ID", "sample_type", "type"))

pseudo_pc %>% head(10) 
```

```{r}
pseudo_pc_mat <- pseudo_pc[, -c(1,2,5)] 

row.names(pseudo_pc_mat) <- paste(pseudo_pc$sample_ID, pseudo_pc$sample_type, pseudo_pc$type, sep = ";")
```
```{r}
#pseudo_pc_mat
cor_ps <- cor(pseudo_pc_mat)[1:2, ] %>% 
  as.data.frame() %>% 
  mutate(PC = c("PC1", "PC2")) %>% 
  select(-PC1, -PC2) %>% 
  gather("cluster", "cor", 1:nrow(temp3 %>% distinct(cluster_name))) %>% 
  mutate(cluster = factor(cluster, levels = c(
    "cluster_1",
    "cluster_2",
    "cluster_3",
    "cluster_4",
    "cluster_5",
    "cluster_6",
    "cluster_7",
    "cluster_8",
    "cluster_9",
    "cluster_10",
    "cluster_11",
    "cluster_12",
    "cluster_13",
    "cluster_14",
    "cluster_15",
    "cluster_16",
    "cluster_17" ))) %>%
  group_by(PC) %>% 
  arrange(-cor) 

 #cor_ps

```
```{r}
cor_ps %>% 
  ggplot(aes(x = PC, y = cluster)) + 
  geom_tile(aes(fill = cor)) +
  scale_fill_gradientn(colours = rev(brewer.pal(11, "Spectral"))) +
   geom_text(aes(label = cor %>% round(3)), size = 5, fontface = "bold") +
  theme_minimal() +
  labs(x = NULL, 
       y = NULL) +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black"))

ggsave(filename = "PC_vs_clusters.svg", width = 4.5, height = 6)
```
```{r}
pal3 <- c("tomato1", "turquoise4", "violetred2")
```

```{r}
ggplot(sex, aes(x = cluster_16, y = cluster_11)) +
  stat_ellipse(aes(fill = sample_type), alpha = 0.3, level = 0.9, type = "t", geom = "polygon") +
  geom_point(aes(color = sample_type), size = 3, alpha = 0.8) + 
  labs(fill = NULL, 
       color = NULL,
       x = paste("logCPM cluster 16, n = ", 
                 temp3 %>%
                   filter(cluster_name == 16 ) %>% 
                   count(), 
                 " miR", sep = ""),  
       y = paste("logCPM cluster 11, n = ", 
                 temp3 %>%
                   filter(cluster_name == 11 ) %>% 
                   count(), 
                 " miR", sep = "")) +
  scale_color_manual(values = pal3) +
  theme_bw() +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 


ggsave(filename = "miR_pseudo_PCR.png")
```

```{r}
logcpm_clust %>% 
  filter(cluster_name == 10) %>% 
  ggplot(aes(x = type, y = z.score)) +
  geom_line(aes(group = miR), size = 2, alpha = 0.8, color = "grey40") +
  geom_point(aes(color = logCPM), size = 3) + 
  facet_wrap(~miR, ncol = 4) +
  ggtitle("cluster 10 - high in egg but low in sperm") + 
  labs(x = NULL,
       color = "log2CPM") +
  scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")) +
  scale_x_discrete(labels = c("egg", "seedling", "sperm")) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 


ggsave(filename = "miR_cluster10.png", width = 8, height = 7, units = "in")
```
```{r}
logcpm_clust %>% 
  filter(cluster_name == 7) %>% 
  ggplot(aes(x = type, y = z.score)) +
  geom_line(aes(group = miR), size = 2, alpha = 0.8, color = "grey40") +
  geom_point(aes(color = logCPM), size = 3) + 
  facet_wrap(~miR, ncol = 4) +
  ggtitle("cluster 7- high in sperm but low in egg") + 
  labs(x = NULL,
       color = "log2CPM") +
  scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")) +
  scale_x_discrete(labels = c("egg", "seedling", "sperm")) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 


ggsave(filename = "miR_cluster7.png", width = 8, height = 7, units = "in")
```



#differential expressions
```{r}
fit <- glmQLFit(y, design, robust = T) 

ov <- glmTreat(fit, contrast = c(-1, 0, 1, 0, 0, 0, 0, 0), lfc=1)

bvc <- glmTreat(fit, contrast = c(0, 0, 0, -1, 0, 0, 0, 1), lfc=1)
```
```{r}
deov <- ov$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(ov$table))

deov %>% head(10)

devc <- bvc$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(bvc$table))

devc %>% head()
```
##visualization on per million miRNA scale
```{r}
a <- DGEList(counts=cts_matrix ,group=sample_des$sample_type)    
a$samples

keep <- rowSums(cpm(a)> 0) >= 0 
a <- a[keep, , keep.lib.sizes = F]
a <- calcNormFactors(a, method = "TMM")
#calcNormFactors calculates normalization factors using TMM
a$samples


design <- model.matrix(~0 + sample_des$sample_type)
a <- estimateDisp(a, design)
#plotBCV(y)
```


```{r}
counts_per_M_miR <- cpm(a, log =F) %>% as.data.frame()
write.csv(cpm, "miR_cpm.csv")
write.csv(counts_per_M_miR, "miR_counts_per_M_miRNA.csv")

 cpm_l <- counts_per_M_miR %>% 
  mutate(miR = row.names(counts_per_M_miR)) %>% 
  gather("sample_ID", "CPM", 1:31) %>% 
  group_by(miR) %>% 
  mutate(z.s = (CPM - mean(CPM))/sd(CPM)) %>%mutate(sample_type = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4" |
      sample_ID == "E9a" |
      sample_ID == "E9b" |
      sample_ID == "E9c" ~ "egg",
    sample_ID == "SP5b" |
      sample_ID == "SP5c" |
      sample_ID == "BS1" |
      sample_ID == "BS2" |
      sample_ID == "BS4a" |
      sample_ID == "BS4b" |
      sample_ID == "BS6a" |
      sample_ID == "BS6b"  ~ "sperm", 
    sample_ID == "BV1" |
      sample_ID == "BV2" ~ "pollen VC", 
    sample_ID == "B15" |
      sample_ID == "B20" |
      sample_ID == "B25" |
      sample_ID == "SD4" ~ "seedling shoot", 
    sample_ID == "P18" |
      sample_ID == "P20" |
      sample_ID == "P25" ~ "sd protoplast", 
    sample_ID == "OV1a" |
      sample_ID == "OV1b" |
      sample_ID == "OV2" ~ "ovary",
    sample_ID == "root" ~ "root",
    sample_ID == "ddm1" |
      sample_ID == "ddm1_con" |
      sample_ID == "drm2" |
      sample_ID == "drm2_con" ~ "leaf"
  )) %>% 
  mutate(type = case_when(
    sample_type == "egg" |
      sample_type == "ovary" ~ "female",
    sample_type == "sperm" |
      sample_type == "pollen VC" ~ "male",
    sample_type == "seedling shoot" | 
      sample_type == "sd protoplast" |
      sample_type == "leaf" |
      sample_type == "root" ~ "vegetative tissues"
  )) %>% 
   mutate(type = factor(type, levels = c("female", "vegetative tissues", "male"))) %>% 
   mutate(genotype = case_when(
     str_detect(sample_ID, "ddm1") &
       str_detect(sample_ID, "_con") == F ~ "ddm1ab",
     str_detect(sample_ID, "drm2")&
       str_detect(sample_ID, "_con") == F ~ "drm2",
     str_detect(sample_ID, "_con")  ~ "WT",
     sample_type != "leaf"  ~ "WT"
   )) %>% 
  as.data.frame()

head(cpm_l)
```
```{r}
deov %>% 
  arrange(-logFC)

deov %>% 
  filter(sig == "s") %>% write_excel_csv("differential expression egg and ovary.csv")
```

```{r}
deov %>% 
  filter(sig == "s") %>% 
  inner_join(cpm_l, by = "miR") %>% 
  filter(type == "female") %>% 
  filter(str_detect(miR, "166|171|399|5519|5790|2863")) %>% 
  mutate(header = substr(miR, start = 5, stop = 100)) %>% 
  ggplot(aes(x = sample_type, y = CPM)) +
  stat_summary(geom = "line", aes(group = miR, color = -log10(FDR)), fun.y = mean, size = 2, alpha = 0.8) +
   stat_summary(geom = "errorbar", fun.data = "mean_cl_boot", width = 0.3) +
  #geom_point(alpha = 0.8) +
  facet_wrap(~ header, scales = "free_y", ncol = 4) +
  labs(x = NULL,
       y = "counts per million miR reads",
       color = "-log10(FDR)") +
  scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")[c(6:9)]) +
  scale_x_discrete(labels = c("egg", "ovary")) +
  theme_minimal() +
  theme(legend.position = "bottom")+
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave(filename = "DE_miR_ov_egg.svg", width = 8, heigh = 4.8) 
```
```{r}
devc %>% 
  filter(sig == "s") %>% 
  inner_join(cpm_l, by = "miR") %>% 
  filter(type == "male") %>% 
  mutate(header = substr(miR, start = 5, stop = 100)) %>% 
  ggplot(aes(x = sample_type, y = CPM)) +
  stat_summary(geom = "line", aes(group = miR, color = -log10(FDR)), fun.y = mean, size = 2, alpha = 0.8) +
  geom_point(alpha = 0.8) +
  facet_wrap(~ header, scales = "free_y") +
  labs(x = NULL,
       y = "counts per million miR reads",
       color = "-log10(FDR)") +
  scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")[c(6:9)]) +
  scale_x_discrete(labels = c("pollen VC", "sperm")) +
  theme_minimal() +
  theme(legend.position = "right" ) +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave(filename = "DE_miR_pveg_sperm.svg", width = 10, heigh = 8) 
```
#miR159 expression
```{r}
miR159_tb <- cpm_l %>% 
  filter(str_detect(miR, "miR159"))

miR159_tb
```
```{r}
#pal
pal5 <- c("tomato1", "orangered3", "orange", "turquoise4", "violetred2")
```


```{r}
miR159_tb %>% 
  #filter(CPM > 100) %>% 
  mutate(class = paste(sample_type, genotype)) %>% 
  filter(sample_type == "leaf" |
           sample_type == "root" |
           sample_type == "seedling shoot") %>%
  ggplot(aes(x = class, y = log10(CPM + 1))) +
  geom_bar(stat = "summary", aes(fill = sample_type), alpha = 0.8, fun.y = mean) +
  #facet_grid(miR ~ sample_type + genotype, scales =  "free", space = "free_x") + 
  facet_wrap(~miR) + 
  geom_point(aes( )) +
  scale_x_discrete(labels = c("ddm1", "drm2", "WT", "root", "seedling")) +
  scale_fill_manual(values = c("darkgreen", "grey40", "turquoise4")) +
  labs(x = NULL,
       y = "log10 counts per million miRNA reads\n",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave(filename = "miR159_leaf_root.png", height = 5, width = 7)
```
##miR159 by total miRNA in each sample type
```{r}
siRNA_comp <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS08/TE/siRNA_comp.xlsx")
```
```{r}
cts_l <- cts %>% 
  gather("sample_ID", "count", 2:32) %>% 
  inner_join(sample_des, by = "sample_ID") 

head(cts_l)
```
```{r}
miR159_tb_s <- cts_l %>%
  filter(str_detect(miR, "miR159")) %>% 
  group_by(sample_ID, sample_type, genotype) %>% 
  summarise(total.count = sum(count)) %>% 
  ungroup() %>% 
  inner_join(siRNA_comp, by = "sample_ID") %>% 
  mutate(proportion = total.count / miRNAs) %>% 
  group_by(sample_type, genotype) %>% 
  summarise(mean.proportion = mean(proportion))

miR159_tb_s

write_excel_csv(miR159_tb_s, "miR159 proprotion out of total miRNA.csv")
```
```{r}
miR159_tb <- cts_l %>%
  filter(str_detect(miR, "miR159")) %>% 
  inner_join(siRNA_comp, by = "sample_ID") %>% 
  mutate(CPM = count / miRNAs * 10^6)

head(miR159_tb)
```

```{r}
miR159_tb %>% 
  #filter(CPM > 100) %>% 
  filter(sample_type != "sd protoplast") %>%
  filter(sample_type != "pollen VC") %>%
  filter(genotype == "WT") %>% 
  mutate(header = substr(miR, start = 5, stop = 100)) %>%
  mutate(class = paste(sample_type, genotype)) %>% 
  mutate(class.f = factor(class, levels = c(
    "leaf WT", "root WT", "seedling shoot WT",
    "ovary WT", "egg WT", "sperm WT"
  ))) %>% 
  ggplot(aes(x = class.f, y = CPM)) +
  geom_bar(stat = "summary", aes(fill = sample_type), alpha = 0.8, fun.y = mean) +
  stat_summary(geom = "errorbar", fun.data = "mean_cl_boot", width = 0.3) +
  #facet_grid(miR ~ sample_type + genotype, scales =  "free", space = "free_x") + 
  facet_wrap(~header, ncol = 2, scales = "free_y") + 
  #geom_point(aes( )) +
  #scale_x_discrete(labels = c("ddm1", "drm2", "", "", "", "", "", "")) +
  scale_x_discrete(labels = NULL) + 
  scale_fill_manual(values = pal) +
  labs(x = NULL,
       y = "counts per million miRNA reads\n",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", size = 18)) +
  theme(axis.text.y=element_text(colour = "black"))  

ggsave(filename = "miR159_gametes.svg", height = 5, width = 3.6)
```
#ath-miR845 in rice
closest match in rice is miR1873, miR1863 and miR5485
```{r}
# cpm_l %>% 
#   filter(str_detect(miR, "miR5485")|
#            str_detect(miR, "miR1863")) %>% 
#   filter(CPM > 10) %>% 
#   ggplot(aes(x = sample_type, y = CPM %>% log10() )) +
#   geom_bar(stat = "summary", aes(fill = sample_type), alpha = 0.8, fun.y = mean) +
#   facet_wrap(~miR, scales =  "free_y") + 
#   geom_point(aes(x = sample_type)) +
#   scale_x_discrete(labels = NULL) +
#   scale_fill_manual(values = pal) +
#   labs(x = NULL,
#        y = "log10 counts per million miRNA reads\n",
#        fill = NULL) +
#   theme_minimal() +
#   theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
#   theme(text = element_text(size = 14, face="bold")) +
#   theme(axis.text.x=element_text(colour = "black")) +
#   theme(axis.text.y=element_text(colour = "black")) +
#   theme(legend.position = "bottom")
# 
# ggsave("putative_miR845_ortho.png")
```

#Egg vs. egg
```{r}
cts_matrix_egg <- cts %>% 
  select(E2, E3, E4, E9a, E9b, E9c)

row.names(cts_matrix_egg) <- cts$miR

sample_des_egg <- sample_des_lib %>% 
  filter(sample_type == "egg") %>% 
  mutate(batch = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4" ~ "batch 1" ,
    str_detect(sample_ID, "9") ~ "batch 2"
  ))

sample_des_egg
```
```{r}
e <- DGEList(counts=cts_matrix_egg ,group=sample_des_egg$batch, lib.size = sample_des_egg$total_smRNA)    
e$samples
```
```{r}
keep <- rowSums(cpm(e)> 1) >= 3 
e <- e[keep, , keep.lib.sizes = T]
e <- calcNormFactors(e, method = "TMM")
#calcNormFactors calculates normalization factors using TMM
e$samples
```
```{r}
design <- model.matrix(~0 + sample_des_egg$batch)
e <- estimateDisp(e, design)
```
```{r}
ecpm <- cpm(e, log =F) %>% as.data.frame()
elogcpm <- cpm(e, log =T) %>% as.data.frame()    #lib.size = sample_des$total_smRNA
#write.csv(ecpm, "egg_gene_siRNA_CPM.csv")
#adonis((t(log(cpm+1))) ~ sample_type * genotype, sample_des, method= "e")
adonis((t(elogcpm)) ~ batch, sample_des_egg, method= "e")
```
```{r}
fitegg <- glmQLFit(e, design, robust = T)
batch_egg <- glmTreat(fitegg, contrast = c(-1, 1), lfc=1)
```
```{r}
deegg <- batch_egg$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(batch_egg$table)) %>% 
  arrange(FDR)

deegg %>% head(10)
```
```{r}
ae <- DGEList(counts=cts_matrix_egg ,group=sample_des_egg$sample_type)    
ae$samples

keep <- rowSums(cpm(ae)> 0) >= 0 
ae <- ae[keep, , keep.lib.sizes = F]
ae <- calcNormFactors(ae, method = "TMM")
#calcNormFactors calculates normalization factors using TMM
ae$samples

design <- model.matrix(~0 + sample_des_egg$batch)
ae <- estimateDisp(ae, design)
```
```{r}
egg_counts_per_M_miR <- cpm(ae, log =F) %>% as.data.frame()
head(egg_counts_per_M_miR)
```

```{r}
ecpm_l <- egg_counts_per_M_miR %>% 
  mutate(miR = row.names(egg_counts_per_M_miR)) %>% 
  gather("sample_ID", "CPM", 1:6) %>% 
  #group_by(miR) %>% 
  #mutate(z.s = (CPM - mean(CPM))/sd(CPM)) %>%
  mutate(sample_type = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4"  ~ "batch 1", 
    sample_ID == "E9a" |
      sample_ID == "E9b" |
      sample_ID == "E9c" ~ "batch 2"
  )) %>% 
  as.data.frame()
head(ecpm_l)
```
```{r}
deegg %>% 
  filter(sig == "s") %>% 
  inner_join(ecpm_l, by = "miR") %>% 
  #filter(type == "female") %>% 
  ggplot(aes(x = sample_type, y = CPM)) +
  stat_summary(geom = "line", aes(group = miR, color = -log10(FDR)), fun.y = mean, size = 2, alpha = 0.8) +
  geom_point(alpha = 0.8) +
  facet_wrap(~ miR, scales = "free_y", ncol = 3) +
  labs(x = NULL,
       y = "counts per million miR reads",
       color = "-log10(FDR)") +
  scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")[c(6:9)]) +
  scale_x_discrete(labels = c("batch 1", "batch 2")) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(strip.text.x = element_text(size = 10)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave(filename = "DE_egg_across_batches.svg", width = 5, heigh = 5) 
```
#phase setting miRNA
```{r}
phase_set_tb <- cts_l %>%
  filter(str_detect(miR, "2118|2275")) %>% 
  inner_join(siRNA_comp, by = "sample_ID") %>% 
  mutate(CPM = count / miRNAs * 10^6)

head(phase_set_tb)
```
```{r}
phase_set_fam <- phase_set_tb %>% 
  mutate(family = case_when(
    str_detect(miR, "2118") ~ "miR2118",
    str_detect(miR, "2275") ~ "miR2275"
  )) %>% 
  group_by(sample_type, family) %>% 
  summarise(tot.CPM = sum(CPM)) %>% 
  ungroup()

head(phase_set_fam, 20)
```

```{r}
phase_set_tb %>% 
  #filter(CPM > 100) %>% 
  filter(sample_type != "sd protoplast") %>%
  #filter(sample_type != "pollen VC") %>%
  filter(genotype == "WT") %>% 
  mutate(header = substr(miR, start = 5, stop = 100)) %>%
  mutate(class = paste(sample_type, genotype)) %>% 
  mutate(class.f = factor(class, levels = c(
    "leaf WT", "root WT", "seedling shoot WT",
    "ovary WT", "egg WT", "sperm WT"
  ))) %>% 
  ggplot(aes(x = class.f, y = CPM)) +
  geom_bar(stat = "summary", aes(fill = sample_type), alpha = 0.8, fun.y = mean) +
  stat_summary(geom = "errorbar", fun.data = "mean_cl_boot", width = 0.3) +
  #facet_grid(miR ~ sample_type + genotype, scales =  "free", space = "free_x") + 
  facet_wrap(~header, ncol = 2, scales = "free_y") + 
  #geom_point(aes( )) +
  #scale_x_discrete(labels = c("ddm1", "drm2", "", "", "", "", "", "")) +
  #scale_x_discrete(labels = NULL) + 
  #scale_fill_manual(values = pal) +
  labs(x = NULL,
       y = "counts per million miRNA reads\n",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", size = 18)) +
  theme(axis.text.y=element_text(colour = "black"))  

#ggsave(filename = "miR159_gametes.svg", height = 5, width = 3.6)
```





Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
